src/sentry/explore/models.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+from typing import Any
+
+from django.db import models, router, transaction
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import FlexibleForeignKey, Model, region_silo_model, sane_repr
+from sentry.db.models.base import DefaultFieldsModel
+from sentry.db.models.fields import JSONField
+from sentry.db.models.fields.bounded import BoundedBigIntegerField, BoundedPositiveIntegerField
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.models.dashboard_widget import TypesClass
+
+
+class ExploreSavedQueryDataset(TypesClass):
+    SPANS = 0
+    OURLOGS = 1
+
+    TYPES = [
+        (SPANS, "spans"),
+        (OURLOGS, "ourlogs"),
+    ]
+    TYPE_NAMES = [t[1] for t in TYPES]
+
+
+@region_silo_model
+class ExploreSavedQueryProject(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    project = FlexibleForeignKey("sentry.Project")
+    explore_saved_query = FlexibleForeignKey("explore.ExploreSavedQuery")
+
+    class Meta:
+        app_label = "explore"
+        db_table = "explore_exploresavedqueryproject"
+        unique_together = (("project", "explore_saved_query"),)
+
+
+@region_silo_model
+class ExploreSavedQuery(DefaultFieldsModel):
+    """
+    A saved Explore query
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
```

Comment:
Having explore query excluded means they will not be part of a self-hosted -> saas migration, or in a region to region relocation. You may want to use Organization scope instead.